app/services/foreman_webhooks/webhook_service.rb

```diff
@@ -38,35 +58,54 @@ def execute
+      uri = URI.parse(url)
 
-      request = http_method.new(uri.request_uri)
-      request.basic_auth(webhook.user, webhook.password) if webhook.user && webhook.password
-      request['Content-Type'] = webhook.http_content_type
+      request = Object.const_get("Net::HTTP::#{http_method.to_s.capitalize}").new(uri.request_uri)
+      request.basic_auth(user, password) if user && password
+      request['Content-Type'] = content_type
       request['X-Request-Id'] = ::Logging.mdc['request'] || SecureRandom.uuid
       request['X-Session-Id'] = ::Logging.mdc['session'] || SecureRandom.uuid
+      headers.each_pair do |key, value|
+        request[key.to_s] = value.to_s
+      end
       request.body = payload
 
-      logger.info("#{webhook.http_method.to_s.upcase} request for webhook #{webhook.name}:")
-      logger.debug("Target: #{uri}")
-      logger.debug("Headers: #{request.to_hash.inspect}")
-      logger.debug("Body: #{request.body.inspect}")
+      Rails.logger.debug("Webhook #{http_method.to_s.upcase} request: #{uri}")
+      Rails.logger.debug("Headers: #{request.to_hash.inspect}")
+      Rails.logger.debug("Body: #{request.body.inspect}")
 
       http = Net::HTTP.new(uri.host, uri.port)
       http.open_timeout = Setting[:proxy_request_timeout]
       http.read_timeout = Setting[:proxy_request_timeout]
       http.ssl_timeout = Setting[:proxy_request_timeout]
       if uri.scheme == 'https'
         http.use_ssl = true
-        http.verify_mode = webhook.verify_ssl? ? OpenSSL::SSL::VERIFY_PEER : OpenSSL::SSL::VERIFY_NONE
-        http.cert_store = webhook.ca_certs_store
+        http.verify_mode = ca_verify ? OpenSSL::SSL::VERIFY_PEER : OpenSSL::SSL::VERIFY_NONE
+        http.cert_store = ca_string if ca_string
+      end
+      http.request(request) do |response|
+        case response
+        when Net::HTTPRedirection then
+          new_location = response['location']
+          Rails.logger.debug "Redirected to #{new_location} (redirects left: #{redirect_limit})"
+          raise(::Foreman::Exception, N_(format('Too many HTTP redirects when calling %{uri}', uri: uri, code: response.code))) if redirect_limit <= 0
+
+          self.request(url: new_location,
+                       payload: payload,
+                       http_method: http_method,
+                       user: user,
+                       password: password,
+                       content_type: content_type,
+                       ca_string: ca_string,
+                       ca_verify: ca_verify,
+                       headers: headers,
+                       follow_redirects: follow_redirects,
+                       redirect_limit: redirect_limit - 1)
+        else
+          response
+        end
       end
-      http.request(request)
     end
   end
```

Comment:
This is a class-method now so it can be possibly used in the future `http_request` helper macro: https://projects.theforeman.org/issues/31299

Also it respects redirects, many services redirect from HTTP to HTTPS.